docs/source/kalman_objective.md

```diff
@@ -0,0 +1,481 @@
+# Kalman-type ensemble approaches
+
+How to think about Kalman-type ensemble based data assimilation:
+
+- What does Kalman-type ensemble methods do?
+- What is their ideal objective?
+- How does the objective behave under statistical estimation?
+- How to think about surrogate objectives?
+- What is the modelling setup and what information is known a-priori?
+- List common methods and discuss them in the context of the above.
+
+What is _not_ provided, is an evaluation-comparison between methods.
+Only how we _think_ they will behave, given what we know about statistics and our objective.
+From the contents, it should however be clear how a simulation study could be constructed to compare methods.
+
+
+## Gaussian conditional transport
+
+Let $x$ and $y$ be random variables jointly multivariate Gaussian with mean and covariance
+
+$$
+\begin{bmatrix}
+x \\
+y
+\end{bmatrix}
+\sim \mathcal{N}\left(\begin{bmatrix}
+\mu_x \\
+\mu_y
+\end{bmatrix}, \begin{bmatrix}
+\Sigma_{x} & \Sigma_{xy} \\
+\Sigma_{yx} & \Sigma_{y}
+\end{bmatrix}\right)
+$$
+
+Define the "Kalman gain" as
+
+$$
+K = \Sigma_{xy}\Sigma_{y}^{-1}.
+$$
+
+Then, a multivariate sample $(x_i,y_i)$ is _transported_ to a sample from the conditional
+$p(x|y)$, having observed $y$ as such, via the formula 
+
+$$
+x_i + K(y-y_i) \sim p(x|y).
+$$
+
+Note that $x|y$ retains Gaussianity as
+
+$$
+x | y \sim \mathcal{N}(\mu_x + K(y - \mu_y), \Sigma_{x} - K \Sigma_{yx}).
+$$
+
+For any distribution having a bijection between (the informative parts of) 
+$x$ and $y$, there exist a similar mapping, transporting a sample $(x,y)$ to $x|y$.
+The EnKF and ES variants follows from writing $y=d=Hx+\epsilon$ where 
+$\epsilon\sim \mathcal{N}(0,\Sigma_{\epsilon})$, thus $\Sigma_y=H\Sigma_xH^T+\Sigma_{\epsilon}$.
+Note that $y$ is common to use as $y=Hx$ but this is not the case in the preliminaries here.
+A point of confusion is that it is indeed _not_ the observation $d$ that is "perturbed" with noise,
```

Comment:
Having mostly read Evensen, I am used to seeing $y = g(x)$ followed by $d=y + \epsilon$ where $f(\epsilon) = \mathcal{N}(0, \Sigma_d)$.
Do I understand correctly that this aligns well with how you think about "perturbed observations"?

As for the term itself, it kinda makes sense to me from a practical point of view.
Users of ert specify an observation $d$ by providing a value and an error $\epsilon$.
To create $D\in\mathcal{1\times N}$, we sample $N$ times from $\mathcal{0, \epsilon}$ and add that to $d$.
In this sense, we "perturbe" $d$ to create $D$.

What would be a better name for $D\in\mathcal{1\times N}$?

Reply:
From a practical point of view, it does not matter. From a theoretical point of view of understanding why these methods work, it matters. (Showcasing how I think about these methods was the goal of the markdown). We need a joint sample from $(x,d)$, and then want to "map" or "transport" this, given an observation say $d^\ast$, to $x|d^\ast$.

To achieve this, a joint sample $(x_i,d_i)$ is needed first. Here $d=h(x)+\epsilon$ or $d=g(x)+\epsilon$ depending on which notation you use. The deterministic model produces $h(x_i)$, but to achieve a correct sample from $d$ we need $d_i=h(x_i)+\epsilon_i$ where $\epsilon_i$ are samples from the observation uncertainty. The innovation is then $d^\ast - d_i$. But, since this is additive, and the noise is additive in $d$, and $\epsilon$ has a symmetric distribution about 0, we may perturb $d^\ast$ instead and compute $d_i^\ast=d^\ast+\epsilon_i$ and see that $d^\ast - d_i$ has the same distribution as $d_i^\ast - g(x_i)$.

The latter term is used extensively, where $d_i^\ast$ are the perturbed observations. In practice this works fine and are distributionally equivalent, but theoretically it adds another layer of confusion to what is really occurring.

---

Note that I do not intend to change language of methods, because I did not intend this as some reference theory document (this is why there are no references, among other things). My goal was to showcase how I personally think. Distribute statistical tools and understanding of methodology given that I think correctly, or become aware of faults in my own logic.

The very particular goal is to create an understanding for LASSO++ to get future PRs with new methodology reviewed and merged. Do we have suggestions/an idea of requirements for such PRs?